tests/eas/load_tracking.py

```diff
@@ -0,0 +1,225 @@
+    frequency. The util_avg value is extracted from scheduler trace events and
+    its mean is compared with the expected value (ignoring the first 300ms so
+    that the signal can stabilize). The test fails if the observed mean is
+    beyond a certain error margin from the expected one. load_avg is then
+    similarly compared with the expected util_avg mean, under the assumption
+    that load_avg should equal util_avg when system load is light.
+
+    Expected Behaviour
+    ==================
+    Load tracking signals are scaled so that the workload results in roughly the
+    same util & load values regardless of frequency.
+    """
+
+    test_conf = {
+        'tools'    : [ 'rt-app' ],
+        'ftrace' : {
+            'events' : [
+                'sched_switch',
+                'sched_load_avg_task',
+                'sched_load_avg_cpu',
+                'sched_pelt_se',
+            ],
+        },
+        'modules': ['cpufreq', 'cgroups'],
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(FreqInvarianceTest, cls).init(*args, **kwargs)
+
+    @memoized
+    @classmethod
+    def _get_cpu(cls, target):
+        # Run on a 'big' CPU, or any CPU if not big.LITTLE
+        if hasattr(target, 'bl'):
+            return target.bl.bigs[0]
+        else:
+            return 0
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        cpu = cls._get_cpu(test_env.target)
+
+        # 10% periodic RTApp workload:
+        wloads = {
+            'fie_10pct' : {
+                'type' : 'rt-app',
+                'conf' : {
+                    'class' : 'periodic',
+                    'params' : {
+                        'duty_cycle_pct': 10,
+                        'duration_s': 3,
+                        'period_ms': 10,
+                    },
+                    'tasks' : 1,
+                    'prefix' : 'fie_test',
+                    'cpus' : [cpu]
+                },
+            },
+        }
+
+        # Create a set of confs with different frequencies
+        # We'll run the 10% workload under each conf (i.e. at each frequency)
+        confs = []
+
+        all_freqs = test_env.target.cpufreq.list_frequencies(cpu)
+        # If we have loads of frequencies just test a cross-section so it
+        # doesn't take all day
+        cls.freqs = all_freqs[::len(all_freqs)/8 + 1]
+        for freq in cls.freqs:
+            confs.append({
+                'tag' : 'freq_{}'.format(freq),
+                'flags' : ['ftrace', 'freeze_userspace'],
+                'cpufreq' : {
+                    'freqs' : {cpu: freq},
+                    'governor' : 'userspace',
+                },
+            })
+
+        return {
+            'wloads': wloads,
+            'confs': confs,
+        }
+
+    def get_expected_util_avg(self, experiment):
+        """
+        Examine trace to figure out an expected mean for util_avg
```

Comment:
It examines the trace via `SchedAssert.getDutyCycle`. 

You're right about the frequency though, it just checks the workload and assumes that all the cpufreq setup worked fine (which I'm OK with, since this isn't testing cpufreq, devlib or LISA). The motivation for that is that we don't currently have the `target.cpufreq.trace_frequencies` functionality working and so I don't expect there to be any cpu_frequency events in the trace.